sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/AzureEventHubsKafkaOAuth2AutoConfiguration.java

```diff
@@ -22,14 +19,12 @@
  * @since 4.3.0
  */
 @Configuration(proxyBeanMethods = false)
-@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
 @ConditionalOnClass(KafkaTemplate.class)
 @ConditionalOnProperty(value = "spring.cloud.azure.eventhubs.kafka.enabled", havingValue = "true", matchIfMissing = true)
 public class AzureEventHubsKafkaOAuth2AutoConfiguration {
 
     @Bean(PASSWORDLESS_KAFKA_PROPERTIES_BEAN_POST_PROCESSOR_BEAN_NAME)
-    @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
-    static BeanPostProcessor kafkaPropertiesBeanPostProcessor(AzureGlobalProperties properties) {
-        return new KafkaPropertiesBeanPostProcessor(properties);
+    static BeanPostProcessor kafkaPropertiesBeanPostProcessor() {
```

Comment:
does this mean there will be two kafka properties bpp in the context?

Reply:
Yes, one for jaas based on conn string, another for jaas based on OAuth2.

Reply:
Are they exclusive? Will they be in the context at the same time?

Reply:
They are logically mutually exclusive, but they do not affect each other; oauth2's BPP will be executed first (postProcessBeforeInitialization), and conn string's BPP will be executed later (postProcessAfterInitialization). When both exist at the same time, oauth2 BPP will not make any changes, and conn string BPP will continue to execute and use conn string for auth.